src/Crypto.res

```diff
@@ -31,6 +31,14 @@ KeyObject = {
   include Impl
 }
 
+module BinaryToTextEncoding = {
+  type t = string
+  let base64: t = "base64"
+  let base64url: t = "base64url"
+  let hex: t = "hex"
+  let binary: t = "binary"
+}
```

Comment:
This does match what the rest of the library does, but I have a mind to start leveraging ReScript's improved binding capability since this project was created.

Something like
```
type binaryToTextEncoding = [ #base64 | #base64url | #hex | #binary ]
```

should work?

Reply:
Yeah, that's how I did it originally but gave up on it since the rest of the library didn't do it that way, but it actually works: [Polymorphic Variant | JavaScript outputs](https://rescript-lang.org/docs/manual/v11.0.0/polymorphic-variant#javascript-output).


But in this scenario I prefer to use variant instead of poly variant because the former is more friendly to prompts:
![image](https://github.com/user-attachments/assets/974b7d61-2bd2-4d74-a5e3-42442d4c0096)

```rescript
module BinaryToTextEncoding = {
  type t = 
   | @as("base64") Base64
   | @as("base64url") Base64Url
   | @as("hex") Hex
   | @as("binary") Binary
}
```

Reply:
What do you think about this? I can refactor similar type definitions in this project with a new PR. ^_^

Reply:
Ah, fair enough. Poly variants can be easier in some cases, but I don't have a strong opinion either way. I think there are some examples of normal variant with `@as` already in the library.

Where the constructors don't conflict, I'd prefer to have it as a `type binaryToTextEncoding` rather than the nested `BinaryToTextEncoding.t` - so it would be similar to my suggestion, but using a normal variant.

Reply:
If you'd like to contribute some patches I'm happy to review. I haven't made a lot of time to maintain this library for a while.